locksmith/src/utils/keyPricer.ts

```diff
@@ -56,6 +57,16 @@ export default class KeyPricer {
     if (lock.keyPrice === '0') {
       return 0
     }
+
+    const { creditCardPrice } = await lockSettingOperations.getSettings({
```

Comment:
Is `keyPriceUSD` the function used when processing actual payments? I want to make sure that if we merge that code, we will actually use the settings' `creditCardPrice` (and not a conversion)

Reply:
this is related for the FE part, i also added the part to pass the correct value for the `purchase` intent for stripe

Reply:
Actually no this is not related to the front-end but to the backend!
We need to make sure that the *same* function on the backend is used to return the price AND to charge that amount.

Reply:
> Actually no this is not related to the front-end but to the backend! We need to make sure that the _same_ function on the backend is used to return the price AND to charge that amount.



> There is a lot going on here. Can you split in smaller PRs that would be more incremental please?
> 
> Maybe the backend changes first thing everything should work the same once they are merged (no lock has a hardcoded price) and then we can look at the front-end changes?

ok, let me split in different PR